CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ColorblindUtilities.java

```diff
@@ -15,100 +15,60 @@
-        surviveFocusChange = true)
-@ActionReferences({
-    @ActionReference(path = "Menu/Display", position = 950, separatorBefore = 901)
-})
-@NbBundle.Messages({
-    "CTL_ApplyColorblindAction=Apply Selected Colorblind Schema to Graph"
-})
-public final class ApplyColorblindAction extends SimpleAction {
-
-    Preferences prefs = NbPreferences.forModule(ApplicationPreferenceKeys.class);
-    String COLORMODE = prefs.get(ApplicationPreferenceKeys.COLORBLIND_MODE, ApplicationPreferenceKeys.COLORBLIND_MODE_DEFAULT);
-
-    public ApplyColorblindAction(GraphNode context) {
-        super(context);
-    }
-
-    @Override
-    protected void edit(PluginGraphs graphs, PluginInteraction interaction, PluginParameters parameters) throws InterruptedException, PluginException {
-        final Graph graph = GraphManager.getDefault().getActiveGraph();
-        final WritableGraph wg = graph.getWritableGraph("Changing colors", true);
-        run(wg);
-        wg.commit();
+    private ColorblindUtilities() {
+        throw new IllegalStateException("Utility class");
     }
 
-    public void run(final WritableGraph graph) {
-        final int vxColorAttr = VisualConcept.VertexAttribute.COLOR.get(graph);
-        final int txColorAttr = VisualConcept.TransactionAttribute.COLOR.get(graph);
+    public static void colorNodes(final GraphWriteMethods wg, final int vxColorblindAttr, final int txColorblindAttr) {
+        Preferences prefs = NbPreferences.forModule(ApplicationPreferenceKeys.class);
+        String COLORMODE = prefs.get(ApplicationPreferenceKeys.COLORBLIND_MODE, ApplicationPreferenceKeys.COLORBLIND_MODE_DEFAULT);
 
-        //Populate the nodes with the colorblind_layer attribute
-        final int vxColorblindAttr = VisualConcept.VertexAttribute.COLORBLIND_LAYER.ensure(graph);
-        final int txColorblindAttr = VisualConcept.TransactionAttribute.COLORBLIND_LAYER.ensure(graph);
+        final int vxColorAttr = VisualConcept.VertexAttribute.COLOR.get(wg);
+        final int txColorAttr = VisualConcept.TransactionAttribute.COLOR.get(wg);
 
-        final int vertexCount = graph.getVertexCount();
-        final int transactionCount = graph.getTransactionCount();
+        final int vertexCount = wg.getVertexCount();
+        final int transactionCount = wg.getTransactionCount();
 
         if (!"None".equals(COLORMODE)) {
             //Iterate through graph vertices. If vertexType is a defined schemaType color will be adjusted if applicable.
             for (int vertex = 0; vertex < vertexCount; vertex++) {
-                final int vxId = graph.getVertex(vertex);
-                final ConstellationColor vertexColor = graph.getObjectValue(vxColorAttr, vxId);
-                final ConstellationColor vxColorblindAlpha = graph.getObjectValue(vxColorblindAttr, vxId);
+                final int vxId = wg.getVertex(vertex);
+                final ConstellationColor vertexColor = wg.getObjectValue(vxColorAttr, vxId);
```

Comment:
I would either add a check for `vxColorAttr != Graph.NOT_FOUND` or use ensure instead of get